packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -2536,10 +2536,14 @@ export class ObjectsInRedisClient {
 
     // -------------- OBJECT FUNCTIONS -------------------------------------------
     // No callback provided by user, we return a Promise
-    private _subscribe(pattern: string, asUser: boolean): Promise<void>;
-    // Callback provided by user, we call callback
-    private _subscribe(pattern: string, asUser: boolean, callback?: ioBroker.ErrorCallback): void;
-    private _subscribe(pattern: string, asUser: boolean, callback?: ioBroker.ErrorCallback): void | Promise<void> {
+    private _subscribe(pattern: string | string[], asUser: boolean): Promise<void>;
+    // Callback provided by user, we call callback|
+    private _subscribe(pattern: string | string[], asUser: boolean, callback?: ioBroker.ErrorCallback): void;
```

Comment:
```suggestion
    private _subscribe(pattern: string | string[], asUser: boolean, callback: ioBroker.ErrorCallback): void;
```
optional (missing) would match the previous overload

Reply:
The problem is that, internally the inference is not working without this type, because we may pass undefined explicitly
![Bildschirmfoto von 2022-05-11 09-11-36](https://user-images.githubusercontent.com/25484431/167790452-eb27ffb2-8c3d-426b-a9d1-4c9164f6f958.png)

Reply:
I can try to solve it via `conditionals` so if `undefined` Promise, else callback